src/ngraph/runtime/gpu/CMakeLists.txt

```diff
@@ -72,8 +72,32 @@ if (NGRAPH_GPU_ENABLE)
 	    -gencode=arch=compute_70,code=compute_70)
     endif()
 
-    set_source_files_properties( ${CUDA_SRC} PROPERTIES CUDA_SOURCE_PROPERTY_FORMAT OBJ)
-    cuda_compile(CUDA_OBJ ${CUDA_SRC} STATIC)
+    set (DO_CUDA_COMPILE FALSE)
+    if (CUDA9_FOUND)
+        set (DO_CUDA_COMPILE TRUE)
+        if (("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU"))
+            # CUDA 9 supports up to gcc 6.x
+            if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0)
+                set (DO_CUDA_COMPILE TRUE)
+            endif()
+        # CUDA 9 supports up to clang 3.9
+        elseif (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.0)
+            set (DO_CUDA_COMPILE TRUE)
+        endif()
+    elseif (("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU"))
```

Comment:
need to check cuda8 found?

Reply:
oh, right.

Reply:
what if you have cuda9 + cuda8 + clang3.8

Reply:
Is it possible have cuda9 + cuda8 + clang3.8 on system?

Reply:
I'm not sure. We look for cuda9 first. Both cuda9 & cuda8 should be good with clang3.8.

Reply:
It is. To start the CMake searches for an optional CUDA 9 install. If it is not found, CUDA 8 is required. Thus, if both 8 and 9 are present CUDA 9 will be used. If clang-3.8 is used in that case, then it will be CUDA 9 + 3.8, which should work without issue.

Reply:
I see, cuda 9 support up to 3.9.